homeassistant/components/wemo/light.py

```diff
@@ -133,27 +126,36 @@ def brightness(self) -> int:
         return cast(int, self.light.state.get("level", 255))
 
     @property
-    def hs_color(self) -> tuple[float, float] | None:
-        """Return the hs color values of this light."""
-        if xy_color := self.light.state.get("color_xy"):
-            return color_util.color_xy_to_hs(*xy_color)
-        return None
+    def xy_color(self) -> tuple[float, float] | None:
+        """Return the xy color value [float, float]."""
+        return self.light.state.get("color_xy")  # type:ignore[no-any-return]
 
     @property
     def color_temp(self) -> int | None:
         """Return the color temperature of this light in mireds."""
         return cast(Optional[int], self.light.state.get("temperature_mireds"))
 
+    @property
+    def color_mode(self) -> ColorMode:
+        """Return the color mode of the light."""
+        if "colorcontrol" in self.light.capabilities:
+            return ColorMode.XY
+        if "colortemperature" in self.light.capabilities:
+            return ColorMode.COLOR_TEMP
+        if "levelcontrol" in self.light.capabilities:
+            return ColorMode.BRIGHTNESS
+        return ColorMode.ONOFF
+
+    @property
+    def supported_color_modes(self) -> set[ColorMode]:
+        """Flag supported color modes."""
+        return {self.color_mode}
```

Comment:
I think this needs to return the set of all ColorModes based on the device capabilities. If I'm understanding the [docs](https://developers.home-assistant.io/docs/core/entity/light/#color-modes) correctly, I think that'd be:

```python
modes: set[ColorMode] = set()
if "colorcontrol" in self.light.capabilities:
    modes.add(ColorMode.HS)
if "colortemperature" in self.light.capabilities:
    modes.add(ColorMode.COLOR_TEMP)
if "levelcontrol" in self.light.capabilities and not modes:
    modes.add(ColorMode.BRIGHTNESS)
if not modes:
    modes.add(ColorMode.ONOFF)
return modes
```